### PR TITLE
fix: skip validation for children

### LINKED ISSUE #151: SkipChildren option not working in validator
SkipChildren only works for root command, not for command path. 
For eg. root command for rhoas-cli is `rhoas`. It works for this, but not for `rhoas kafka` (contains ..list, ..create etc)

### PR DESCRIPTION
**Pull request by ankithans**
Closes #151 

### Description
- fix skip children commands by specifying path

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by jjkiely** [comment]
Have you verified that this fixes the issue? If not can you provide some steps I can follow in order to do so? Thanks!

File: validator/rules/cmd_test.go [hunk 1]
```diff
@@ -13,24 +12,19 @@ func Test_ExecuteCommand(t *testing.T) {
 	// default config can also be overrided
 	ruleCfg := ValidatorConfig{
 		ValidatorOptions: ValidatorOptions{
-			SkipCommands: map[string]bool{"cmd100 cmd0 subcmd01": true},
```
Comment (wtrocki) [inline validator/rules/cmd_test.go:16]: Is this removed?

**Comment by ankithans** [comment]
yep verified myself.
steps are:
- run `make test/validator` or `go run ./validator/rules`
- change the stuff in `validator/rules/cmd_test.go` 

```go
ValidatorOptions: ValidatorOptions{
	SkipChildren: map[string]bool{
		"root echo": true,
	},
}
```
this will stop checking root echo and root echo .... children

File: validator/rules/cmd_test.go (hunk 1 above)
Comment (ankithans) [reply validator/rules/cmd_test.go:16]: no

Comment (ankithans) [reply validator/rules/cmd_test.go:16]: just removed from test

**Comment by wtrocki** [comment]
This is breaking change also removes other flags that we have documented. Best to put more info into PR.

I would like if we can avoid having multiple flags for skipping

Skip for children can be done by 
```
CMD data manipulators *
```

Where last `*` means children only.
I'm little bit like broken record but..

lets start with docs and api description

**Comment by ankithans** [comment]
> Skip for children can be done by
CMD data manipulators *

got it. this looks cool and is easy to use. Let's do this.

> I'm little bit like broken record but..
> lets start with docs and api description

you telling about skipping issue or the whole project in general

**Comment by wtrocki** [comment]
> you telling about skipping issue or the whole project in general.

No. I meant that I will not review PRs without docs :)

**Comment by ankithans** [comment]
oh okay, if you haven't noticed, this was not a breaking change and docs was already there : https://github.com/aerogear/charmil/blob/validator-skip/docs/src/validator.md#ignore-commands

but now let's do the proposed change and will update the docs.

**Comment by ankithans** [comment]
docs: https://github.com/aerogear/charmil/blob/validator-skip/docs/src/validator.md#ignore-commands

**Comment by ankithans** [comment]
yep, we have a test using SkipCommands attribute

File: validator/rules/executor.go [hunk 2]
```diff
@@ -70,6 +72,14 @@ func executeRecursive(cmd *cobra.Command, info *validator.StatusLog, ruleConfig
 
 // executeRulesChildren execute rules on children of cmd
 func executeRulesChildren(cmd *cobra.Command, info *validator.StatusLog, ruleConfig *RuleConfig, userValidatorConfig *ValidatorConfig) []validator.ValidationError {
+
+	// if command's children needs to be ignored
+	if val, ok := userValidatorConfig.ValidatorOptions.SkipCommands[cmd.CommandPath()+"*"]; ok {
```
Comment (wtrocki) [inline validator/rules/executor.go:77]: This should be helper method

**Review by wtrocki — APPROVED** [review]

**Comment by wtrocki** [comment]
> but now let's do the proposed change and will update the docs.

Docs first please.

**Comment by ankithans** [comment]
> Docs first please next time.

yess 💪

File: validator/rules/executor.go (hunk 2 above)
Comment (ankithans) [reply validator/rules/executor.go]: done

**Comment by wtrocki** [comment]
We can release now

**Pull request merged**